feat(theme): move aalborg theme into core

hypeJunction opened the pull request:
BREAKING CHANGE
Aalborg theme has been distributed among core views and plugins.
This may affect plugins that have relied on core styles,
or a certain priority of views

hypeJunction commented:
Eh, how are these conflicts possible?

jdalsem commented on docs/guides/upgrading.rst:
```diff
@@ -237,6 +237,18 @@ required to make the old interface work.
+
+Aalborg theme is no longer bundled with Elgg, instead all styles have been either moved to core views or to an appropriate core plugin.
+
+This change may affect themes and plugins that were developed without aalboarg stylesheets loaded or were relying on a certain priority of views with aalborg stylesheets loaded.
```
Aalborg typo

jdalsem commented on engine/lib/admin.php:
```diff
@@ -314,9 +314,10 @@ function _elgg_admin_init() {
+			'text' => elgg_echo('admin'),
+			'priority' => 101,
 			'section' => 'alt',
 		));
```
i was wondering if it makes more sense to move this to the users_pagesetup? as it relies on the account parent...

jdalsem commented on engine/lib/users.php:
```diff
@@ -764,20 +764,32 @@ function users_pagesetup() {
-			'text' => elgg_view_icon('settings') . elgg_echo('settings'),
-			'priority' => 500,
+			'text' => elgg_echo('settings'),
+			'priority' => 103,
```
can we have a bit more room in priorities? it is pretty hard to put a menu item between them in this way... just make them 150 and 200 or something

jdalsem commented on views/default/elements/buttons.css.php:
```diff
@@ -1,147 +0,0 @@
-	text-shadow: 1px 1px 0px black;
-	text-decoration: none;
-	border: 1px solid #4690d6;
-	background: #4690d6 url(<?= elgg_get_simplecache_url("button_graduation.png"); ?>) repeat-x left 10px;
```
button_graduation image can also be removed

hypeJunction replied:
This is the old file. No longer there.

jdalsem replied:
i meant the image file in core _graphics could also be removed

jdalsem commented on views/default/elements/buttons.css.php:
```diff
@@ -1,147 +0,0 @@
-
-/* Action: This button should convey a normal, inconsequential action, such as clicking a link */
-.elgg-button-action {
-	background: #ccc url(<?= elgg_get_simplecache_url("button_background.gif"); ?>) repeat-x 0 0;
```
button_background as an image can be removed

jdalsem commented on views/default/elements/forms.css.php:
```diff
@@ -1,81 +1,79 @@
-<?php
+/* <style> /**/
```
can become a normal css file (if you change the image url)

hypeJunction replied:
We will make it a normal css after loosing friendspicker.

jdalsem commented on views/default/page/elements/header.php:
```diff
@@ -8,7 +8,7 @@
 
 // drop-down login
-echo elgg_view('core/account/login_dropdown');
+//echo elgg_view('core/account/login_dropdown');
```
why not remove this?

jdalsem commented on views/default/page/elements/header.php:
```diff
@@ -8,7 +8,7 @@
 // insert site-wide navigation
-echo elgg_view_menu('site');
\ No newline at end of file
+//echo elgg_view_menu('site');
```
remove?

jdalsem commented:
Could you check if we can move some .php css files to normal css while at it?

i also was hoping that we could separate the media queries in separate views

hypeJunction commented:
I prefer media queries closer to elements they overwrite.

jdalsem commented:
the downside is that theming require you to copy all responsive behaviour...

the current way is that the media queries are at the bottom of the files... not super close to the elements they change... we could split it into 2 views...
buttons.css + buttons_responsive.css etc

hypeJunction commented:
I hope we will throw most of them out eventually.

jdalsem commented:
if you can resolve the conflicts and address my comments... i can give my thumbsup

jdalsem commented:
don't want this lying around to long... (also conflicting with my walled garden PR), so this should go in before that PR

hypeJunction commented:
I had to rebuild git index to resolve conflicts, but that picked up all the files that had bad chmod.

I don't want to rename all the .php files to .css just yet, because I want to break them down into smaller parts, and I think we will end up renaming them back to .php.

I changed the priorities of topbar items.

hypeJunction committed:
feat(theme): move aalboarg theme into core

Moves aalborg theme into core and plugins.
Rebuilds git index cache.

BREAKING CHANGE
Aalborg theme has been moved into core and bundled plugins.
Third-party plugins that have been developed without aalborg
stylesheets loaded, or relying on a certain view priorities
maybe affected.
Topbar menu item registration has changed: some items have a
new parent item, some have been moved to a different section,
some have a new priority.

hypeJunction commented:
Meh.
